Assets/MixedRealityToolkit/Boundary/MixedRealityBoundarySystem.cs

```diff
@@ -0,0 +1,235 @@
+
+            if (Boundary.TryGetGeometry(boundaryGeometry, Boundary.Type.TrackedArea))
+            {
+                // FloorHeight starts out as null. Use a suitably high value for the floor to ensure
+                // that we do not accidentally set it too low.
+                float floorHeight = 10000f;
+
+                for (int i = 0; i < boundaryGeometry.Count; i++)
+                {
+                    Vector3 pointA = boundaryGeometry[i];
+                    Vector3 pointB = boundaryGeometry[(i + 1) % boundaryGeometry.Count];
+                    boundaryEdges.Add(new Edge(pointA, pointB));
+
+                    floorHeight = Mathf.Min(floorHeight, boundaryGeometry[i].y); 
+                }
+
+                FloorHeight = floorHeight;
+                Bounds = boundaryEdges.ToArray();
+                CreateInscribedBounds();
+            }
+            else
+            {
+                // TODO: How do we determine if we have a floor without boundaries
+                Debug.LogWarning("Failed to calculate boundary bounds.");
+            }
+        }
+
+        /// <summary>
+        /// Creates the two dimensional volume described by the largest rectangle that
+        /// is contained withing the playspace geometry and the configured height.
+        /// </summary>
+        private void CreateInscribedBounds()
+        {
+            // We always use the same seed so that from run to run, the inscribed bounds are consistent.
+            rectangularBounds = new InscribedRectangle(Bounds, Mathf.Abs("Mixed Reality Toolkit".GetHashCode()));
+        }
+
+        /// <summary>
+        /// Updates the <see cref="TrackingSpaceType"/> on the XR device.
+        /// </summary>
+        private void SetTrackingSpace()
+        {
+            TrackingSpaceType trackingSpace;
+
+            // In current versions of Unity, there are two types of tracking spaces. For boundaries, if the scale
+            // is not Room or Standing, it currently maps to TrackingSpaceType.Stationary.
+            switch (Scale)
+            {
+                case ExperienceScale.Standing:
+                case ExperienceScale.Room:
+                    trackingSpace = TrackingSpaceType.RoomScale;
+                    break;
+
+                case ExperienceScale.OrientationOnly:
+                case ExperienceScale.Seated:
+                case ExperienceScale.World:
+                    trackingSpace = TrackingSpaceType.Stationary;
+                    break;
+
+                default:
+                    trackingSpace = TrackingSpaceType.Stationary;
+                    Debug.LogWarning("Unknown / unsupported ExperienceScale. Defaulting to Stationary tracking space.");
+                    break;
+            }
+
+            bool trackingSpaceSet = XRDevice.SetTrackingSpaceType(trackingSpace);
+            if (!trackingSpaceSet)
+            {
+                // TODO: Implement ExperienceScale fallback logic
```

Comment:
Can we add a comment to the header of this class to detail it's capabilities, if some are not intended to be implemented until later.
Also ensure the boundary documentation reflects this.

Reply:
Reworded the todo. You are correct that we need to properly document / expose the proposed fallback. Hopefully this will pass muster :)

            if (!trackingSpaceSet)
            {
                // TODO: consider how best to handle this situation
            }